### PR TITLE
App cannot be removed if the creator is deleted

### PR DESCRIPTION
**Pull request by loganhz**
Problem:
Cannot delete Rancher catalog apps when their creator user is deleted

Solution:
We should fallback to use project token to remove helm

https://github.com/rancher/rancher/issues/16274

### CODE REVIEW

**Comment by StrongMonkey** [comment]
@cjellick to give you the context, before this change we bind each catalog app with the creator ID so that app is deployed with whatever the permission comes with the creator. This introduced a bug that if creator is deleted further operations can't be continued. So we are thinking to fallback to project-admin permission for catalog app which creator has been deleted. This workaround will introduce a new issue where any user in this project are able to edit this app can use project-admin permission to deploy stuff.

**Comment by cjellick** [comment]
The problem that @StrongMonkey points out represents a privilege escalation issue. We cannot merge without addressing that. Here is my proposed solution:
if you get a is-not-found error on the user (as @StrongMonkey suggests), then check if the current user has permission to create apps (can use the subject access review api for this) and if they do, set them as the creator (effectively transfering ownership) and use them to execute the helm task

**Comment by loganhz** [comment]
@cjellick @StrongMonkey I would like to address the upgrade issue https://github.com/rancher/rancher/issues/17265  in another PR.

In this PR, it will only fallback when users remove the app.

File: pkg/controllers/user/helm/controller.go [hunk 1]
```diff
@@ -272,12 +275,15 @@ func (l *Lifecycle) createAppRevision(obj *v3.App, template, notes string, faile
 }
 
 func (l *Lifecycle) writeKubeConfig(obj *v3.App, tempDir string) (string, error) {
+	var token string
+
 	userID := obj.Annotations["field.cattle.io/creatorId"]
 	user, err := l.UserClient.Get(userID, metav1.GetOptions{})
 	if err != nil {
```
[the remark this answers is not in the document]
Comment (loganhz) [reply pkg/controllers/user/helm/controller.go]: Updated. Thanks!

**Comment by cjellick** [comment]
LGTM

**Pull request merged**